src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c

```diff
@@ -748,15 +748,16 @@ int xocl_sync_bo_ioctl(struct drm_device *dev,
 
 	xobj = to_xocl_bo(gem_obj);
 	BO_ENTER("xobj %p", xobj);
-	sgt = xobj->sgt;
-	sg = sgt->sgl;
 
-	if (!xocl_bo_sync_able(xobj->flags)) {
+	if (!xocl_bo_sync_able(xobj->flags) || !xobj->sgt) {
```

Comment:
Wondering if we don't have to change this line? Basically, checking !xocl_bo_sync_able(xobj->flags) might be good enough... It makes sense to first do the !xocl_bo_sync_able(xobj->flags) check before start doing the real job.